File: 2197-decode-the-slanted-ciphertext/decode-the-slanted-ciphertext.py
# lee215 solution

class Solution:
    def decodeCiphertext(self, encodedText: str, rows: int) -> str:
        cols = len(encodedText)//rows
        res = []
        for i in range(cols):
            for j in range(i, len(encodedText), cols + 1):
                res.append(encodedText[j])
        return "".join(res).rstrip()

File: 2197-decode-the-slanted-ciphertext/decode-the-slanted-ciphertext.java
class Solution {
    public String decodeCiphertext(String encodedText, int rows) {
        int length = encodedText.length();
        int cols = length / rows;
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < cols; i++) {
            for (int j = i; j < length; j += cols + 1) {
                sb.append(encodedText.charAt(j));
                // if (j == length - 1) return sb.toString();
            }
        }

        return sb.toString().stripTrailing();
    }
}